components/app_update/include/esp_ota_ops.h

```diff
@@ -116,6 +116,8 @@ esp_err_t esp_ota_begin(const esp_partition_t* partition, size_t image_size, esp
  *    - ESP_ERR_OTA_VALIDATE_FAILED: First byte of image contains invalid app image magic byte.
  *    - ESP_ERR_FLASH_OP_TIMEOUT or ESP_ERR_FLASH_OP_FAIL: Flash write failed.
  *    - ESP_ERR_OTA_SELECT_INFO_INVALID: OTA data partition has invalid contents
+ *    - ESP_ERR_INVALID_SIZE: if write would go out of bounds of the partition
+ *    - ESP_ERR_NOT_ALLOWED: if partition is read-only
```

Comment:
`esp_ota_write()` will never return `ESP_ERR_NOT_ALLOWED` as the partitions of subtype `app` are never read-only. For reference: https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/partition-tables.html#flags. I think we can remove this.

I think we can just add the change below as a placeholder for other flash related error codes:
```suggestion
 *    - or one of error codes from lower-level flash driver.
```